hathor/p2p/states/ready.py

```diff
@@ -180,3 +200,48 @@ def handle_pong(self, payload: str) -> None:
         self.ping_min_rtt = min(self.ping_min_rtt, self.ping_rtt)
         self.ping_start_time = None
         self.log.debug('rtt updated', rtt=self.ping_rtt, min_rtt=self.ping_min_rtt)
+
+    def send_get_best_blockchain(self, nBlocks: str = '10') -> None:
+        """ Send a GET-BEST-BLOCKCHAIN command, requesting a list of latest
+        N blocks from the best blockchain.
+        """
+        self.send_message(ProtocolMessages.GET_BEST_BLOCKCHAIN, nBlocks)
+
+    def handle_get_best_blockchain(self, payload: str) -> None:
+        """ Executed when a GET-BEST-BLOCKCHAIN command is received.
+        It just responds with a list N blocks from the best blockchain.
+        """
+        nBlocks = int(payload)
+        if not (0 < nBlocks < 101):
+            self.protocol.send_error_and_close_connection('Invalid N to get best blockchain.')
+            return
+
+        best_blockchain: list[BlockInfo] = []
+        tx_storage = self.protocol.node.tx_storage
+        block = tx_storage.get_best_block()
+
+        while len(best_blockchain) < nBlocks:
+            block_info = BlockInfo(
+                block.hash_hex,
+                block.get_height(),
+                block.weight)
+            best_blockchain.append(block_info)
+            # halt if genesis block
+            if block_info.height == 0:
+                break
+            block = block.get_block_parent()
+
+        self.send_best_blockchain(best_blockchain)
+
+    def send_best_blockchain(self, best_blockchain: list[BlockInfo]) -> None:
+        """ Send a BEST-BLOCKCHAIN command with a best blockchain of N blocks.
+        """
+        self.send_message(ProtocolMessages.BEST_BLOCKCHAIN, json_dumps(best_blockchain))
+
+    def handle_best_blockchain(self, payload: str) -> None:
+        """ Executed when a BEST-BLOCKCHAIN command is received. It updates
+        the best blockchain.
+        """
+        restored_blocks = json_loads(payload)
+        best_blockchain = [BlockInfo(*block) for block in restored_blocks]
```

Comment:
Maybe rename `block` to `block_info_raw`? Don't you have to validate the types are correct?

Reply:
Done!